src/service/world-api-adapter.js

```diff
@@ -513,13 +515,14 @@ module.exports = function (config) {
         *  **Example**: This method is not yet implemented.
         *
         *  **Parameters**
-        * @param {string} `worldId`
+        * @param {object} `params` worldId and model should be provided.
         * @param {object} `options` (Optional) Options object to override global options.
         */
-        newRunForWorld: function (worldId, options) {
+        newRunForWorld: function (params, options) {
+            var worldId = params.worldId;
             return this.deleteRun(worldId)
                 .then(function () {
-                    return this.getCurrentRunId({ filter: worldId });
+                    return this.getCurrentRunId({ model: params.model, filter: worldId });
```

Comment:
we should merge everything before getting the model name from the options|params